craft_parts/overlays/overlay_fs.py

```diff
@@ -0,0 +1,115 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Low level interface to OS overlayfs."""
+
+import logging
+import os
+from pathlib import Path
+from subprocess import CalledProcessError
+from typing import List, Optional, Union
+
+from craft_parts.utils import os_utils
+
+from . import errors
+
+logger = logging.getLogger(__name__)
+
+
+class OverlayFS:
+    """Linux overlayfs operations."""
+
+    def __init__(
+        self, *, lower_dir: Union[Path, List[Path]], upper_dir: Path, work_dir: Path
+    ):
+        if not isinstance(lower_dir, list):
```

Comment:
Consider making it `lower_dirs` and always require a list so this isn't responsible for the conversion (and to keep the interface a touch simpler).

Reply:
Changed to always receive a list.